tests/vmi_networking_test.go

```diff
@@ -447,4 +447,26 @@ var _ = Describe("Networking", func() {
 			checkMacAddress(deadbeafVMI, deadbeafVMI.Spec.Domain.Devices.Interfaces[0].MacAddress, "localhost:~#")
 		})
 	})
+
+	Context("VirtualMachineInstance with disabled automatic attachment of interfaces", func() {
+		It("should not configure any external interfaces", func() {
+			By("checking loopback is the only guest interface")
+			autoAttach := false
+			detachedVMI := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			detachedVMI.Spec.Domain.Devices.AutoAttachPodInterface = &autoAttach
+
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(detachedVMI)
+			Expect(err).ToNot(HaveOccurred())
+			waitUntilVMIReady(detachedVMI, tests.LoggedInCirrosExpecter)
+
+			err := tests.CheckForTextExpecter(detachedVMI, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "ls /sys/class/net/\n"},
+				&expect.BExp{R: "lo"},
+				&expect.BExp{R: "\\$ "},
```

Comment:
I think that this test would also succeed if other network devices are present for as long as "lo" would appear in the list.

Reply:
You are right, I didn't expect it to silently "skip" eth0 entry.